src/_pytest/config/findpaths.py

```diff
@@ -161,7 +161,7 @@ def determine_setup(
     args: List[str],
     rootdir_cmd_arg: Optional[str] = None,
     config: Optional["Config"] = None,
-) -> Tuple[py.path.local, Optional[str], Dict[str, Union[str, List[str]]]]:
+) -> Tuple[py.path.local, Optional[py.path.local], Dict[str, Union[str, List[str]]]]:
```

Comment:
Please make sure to convert `inifile` to a `py.path.local` inside `if inifile:` below to honor this type annotation.

Reply:
I have a PR locally which addresses these issues (by fixing errors after running mypy with https://github.com/pytest-dev/py/pull/232). I'll post it in a bit.